core/src/main/java/hudson/slaves/JNLPLauncher.java

```diff
@@ -221,4 +221,18 @@ public boolean filterType(@Nonnull Class<?> contextClass, @Nonnull Descriptor de
         }
     }
 
+    /**
+     * Returns true if Java Web Start button should be displayed.
+     * Java Web Start is only supported when the Jenkins server is
+     * running with Java 8.  Earlier Java versions are not supported by Jenkins.
+     * Later Java versions do not support Java Web Start.
+     *
+     * This flag is checked in {@code config.jelly} before displaying the
+     * Java Web Start button.
+     * @return {@code true} if Java Web Start button should be displayed.
+     * @since FIXME
+     */
```

Comment:
Add `@Restricted(NoExternalUse.class)`? We are about moving Java Web Start codebase to a plugin, so a restriction would be helpful to limit the method exposure

Reply:
Done